opentelemetry-sdk/src/logs/growable_array.rs

```diff
@@ -0,0 +1,369 @@
+
+        assert_eq!(
+            collection.get(0).map(|kv| (&kv.0, &kv.1)),
+            Some((&key1, &value1))
+        );
+        assert_eq!(
+            collection.get(1).map(|kv| (&kv.0, &kv.1)),
+            Some((&key2, &value2))
+        );
+        assert_eq!(collection.len(), 2);
+
+        // Test iterating over the key-value pairs
+        let mut iter = collection.into_iter();
+        assert_eq!(iter.next(), Some(KeyValuePair(key1, value1)));
+        assert_eq!(iter.next(), Some(KeyValuePair(key2, value2)));
+        assert_eq!(iter.next(), None);
+    }
+
+    #[test]
+    fn test_empty_attributes() {
+        let collection = GrowableArray::<KeyValuePair>::new();
+        assert_eq!(collection.len(), 0);
+        assert_eq!(collection.get(0), None);
+
+        let mut iter = collection.into_iter();
+        assert_eq!(iter.next(), None);
+    }
+
+    #[test]
+    fn test_less_than_max_stack_capacity() {
+        let mut collection = GrowableArray::<i32>::new();
+        for i in 0..DEFAULT_MAX_STACK_CAPACITY - 1 {
+            collection.push(i as i32);
+        }
+        assert_eq!(collection.len(), DEFAULT_MAX_STACK_CAPACITY - 1);
+
+        for i in 0..DEFAULT_MAX_STACK_CAPACITY - 1 {
+            assert_eq!(collection.get(i), Some(&(i as i32)));
+        }
+        assert_eq!(collection.get(DEFAULT_MAX_STACK_CAPACITY), None);
+
+        let mut iter = collection.into_iter();
+        for i in 0..DEFAULT_MAX_STACK_CAPACITY - 1 {
+            assert_eq!(iter.next(), Some(i as i32));
+        }
+        assert_eq!(iter.next(), None);
+    }
+
+    #[test]
+    fn test_exactly_max_stack_capacity() {
+        let mut collection = GrowableArray::<i32>::new();
+        for i in 0..DEFAULT_MAX_STACK_CAPACITY {
+            collection.push(i as i32);
+        }
+        assert_eq!(collection.len(), DEFAULT_MAX_STACK_CAPACITY);
+
+        for i in 0..DEFAULT_MAX_STACK_CAPACITY {
+            assert_eq!(collection.get(i), Some(&(i as i32)));
+        }
+        assert_eq!(collection.get(DEFAULT_MAX_STACK_CAPACITY), None);
+
+        let mut iter = collection.into_iter();
+        for i in 0..DEFAULT_MAX_STACK_CAPACITY {
+            assert_eq!(iter.next(), Some(i as i32));
+        }
+        assert_eq!(iter.next(), None);
+    }
+
+    #[test]
+    fn test_exceeds_stack_but_not_initial_vec_capacity() {
+        let mut collection = GrowableArray::<i32>::new();
+        for i in 0..(DEFAULT_MAX_STACK_CAPACITY + DEFAULT_INITIAL_VEC_CAPACITY - 1) {
+            collection.push(i as i32);
+        }
+        assert_eq!(
+            collection.len(),
+            DEFAULT_MAX_STACK_CAPACITY + DEFAULT_INITIAL_VEC_CAPACITY - 1
+        );
+
+        for i in 0..(DEFAULT_MAX_STACK_CAPACITY + DEFAULT_INITIAL_VEC_CAPACITY - 1) {
+            assert_eq!(collection.get(i), Some(&(i as i32)));
+        }
+        assert_eq!(
+            collection.get(DEFAULT_MAX_STACK_CAPACITY + DEFAULT_INITIAL_VEC_CAPACITY),
```

Comment:
Same as this [comment](https://github.com/open-telemetry/opentelemetry-rust/pull/1965/files#r1698884942).